contracts/cw721-base/src/execute.rs

```diff
@@ -76,11 +79,11 @@ where
     }
 }
 
-// TODO pull this into some sort of trait extension??
-impl<'a, T, C, E, Q> Cw721Contract<'a, T, C, E, Q>
+impl<'a, T, C, I, E, Q> Cw721Contract<'a, T, C, I, E, Q>
 where
     T: Serialize + DeserializeOwned + Clone,
     C: CustomMsg,
+    I: CustomMsg + DeserializeOwned,
     E: CustomMsg,
     Q: CustomMsg,
```

Comment:
Might be nice to give these full type names because there are so many of them:

```suggestion
    MintExt: Serialize + DeserializeOwned + Clone,
    ResponseExt: CustomMsg,
    InstantiateExt: CustomMsg + DeserializeOwned,
    ExecuteExt: CustomMsg,
    QueryExt: CustomMsg,
```

Takes some digging around to read otherwise.

Reply:
Definitely, will do.